packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -0,0 +1,734 @@
+import * as codebuild from '@aws-cdk/aws-codebuild';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
```

Comment:
this file feels really large, can we break it down?
would any of this apply for `StopBuild` - if so we should start moving shared types out. 

take a look at the `EcsRunTask` service integration for inspiration on how to break this down